debian/postinst

```diff
@@ -29,8 +29,17 @@ insert_nss_entry() {
 }
 
 action="$1"
+previous_version="$2"
 
 if [ configure = "$action" ]; then
     pam-auth-update --package
     insert_nss_entry
+
+    # We installed /etc/authd with permissions 777 - umask in version 0.3.7 and earlier
+    if dpkg --compare-versions "$previous_version" le 0.3.7; then
+        # Ensure that the /etc/authd directory has mode 700
+        if [ -d /etc/authd ] && [ "$(stat -c %a /etc/authd)" != "700" ]; then
+            chmod 700 /etc/authd
```

Comment:
Nice, should we also set `/etc/authd/brokers.d` and their files to 600?

Reply:
We don't create that, it's created by the user, so I would prefer keeping it the way the user created it. We could change [the instructions](https://github.com/ubuntu/authd/blob/f7fb3f6280d9edd6ecd5df6d5831d151d0e674f1/docs/howto/configure-authd.md#L8-L9) to create them with 700/600.